src/installer/pkg/sfx/Microsoft.NETCore.App/ReadyToRun.targets

```diff
@@ -48,7 +49,7 @@
   <Target Name="PatchDotnetRootBeforeRunningCrossgen2" BeforeTargets="_CreateR2RImages">
     <PropertyGroup>
       <OriginalDotnetRootValue>$(DOTNET_ROOT)</OriginalDotnetRootValue>
-      <DOTNET_ROOT>$(RepoRoot)</DOTNET_ROOT>
+      <DOTNET_ROOT>$([System.IO.Path]::Combine('$(RepoRoot)', '.dotnet'))</DOTNET_ROOT>
```

Comment:
This isn't reliable. The dotnet.cmd tool will not always point at the .dotnet directory.  Instead you need to read the directory from artifacts/toolset/sdk.txt.

Reply:
Fixed in 4th commit. I have also added a temporary instrumentation showing the <code>DOTNET_ROOT</code> location as the installer builds for some reason still fail to find the correct SDK even though I believe lab machines use the default <code>.dotnet</code> location.